code/_onclick/hud/spell_screen_objects.dm

```diff
@@ -144,7 +144,7 @@
 	..()
 	if(spellmaster)
 		spellmaster.spell_objects -= src
-	if(spellmaster && !spellmaster.spell_objects.len)
+	if(spellmaster && !spellmaster.spell_objects.len && master)
```

Comment:
Someone tell me why this isn't the ideal solution. Seriously, please do. I'm so confused.